hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/clustering/run/strategy/MultipleSparkJobExecutionStrategy.java

```diff
@@ -457,9 +457,10 @@ private Dataset<Row> readRecordsForGroupAsRow(JavaSparkContext jsc,
 
     String readPathString =
         String.join(",", Arrays.stream(paths).map(StoragePath::toString).toArray(String[]::new));
+    String globPathString = String.join(",", Arrays.stream(paths).map(StoragePath::getParent).map(StoragePath::toString).distinct().toArray(String[]::new));
     params.put("hoodie.datasource.read.paths", readPathString);
     // Building HoodieFileIndex needs this param to decide query path
-    params.put("glob.paths", readPathString);
+    params.put("glob.paths", globPathString);
 
```

Comment:
do we have any test cases?

Reply:
I can't find a test class matching this class name. Is there a clustering test suite I should look in?

Reply:
Not sure whether `TestHoodieSparkMergeOnReadTableClustering` is the candidate.

Reply:
Looks like it is already covered there

Reply:
Fine, let's merge it first.